pkg/R/DataFrame.R

```diff
@@ -52,21 +53,108 @@ dataFrame <- function(sdf) {
 #' printSchema(df)
 #'}
 
-setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+setGeneric("printSchema", function(x) { standardGeneric("printSchema") })
 
 setMethod("printSchema",
-          signature(df = "DataFrame"),
-          function(df) {
-            sdf <- df@sdf
-            schemaString <- callJMethod(sdf, "printSchema")
+          signature(x = "DataFrame"),
+          function(x) {
+            schemaString <- callJMethod(x@sdf, "printSchema")
             cat(schemaString)
           })
 
+#' Get schema object
+#' 
+#' Returns the schema of this DataFrame as a list of named lists. Each named
```

Comment:
@cafreeman I am not sure I completely understand the options here. I can see that PySpark implemented pure Python objects for every data type in SparkSQL and there is a lot of code for that which might be good to avoid.

The other way seems to me is to have a StructType S4 class in R which holds a `jobj` to a StructType class in JVM. (Similar to say what we do with DataFrame etc.) What are the methods we want to support on `StructType` ? Will it just be a `print`which can be used to print schemas ? I guess we might add methods to construct StructTypes in the future.
@davies is this what you had in mind ?